qiskit_optimization/applications/bin_packing.py

```diff
@@ -0,0 +1,97 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the set packing."""
+
+from typing import List, Union
+
+import numpy as np
+from docplex.mp.model import Model
+
+from qiskit_optimization.algorithms import OptimizationResult
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+from qiskit_optimization.translators import from_docplex_mp
+from .optimization_application import OptimizationApplication
+
+
+class BinPacking(OptimizationApplication):
+    """Optimization application for the "bin packing" [1] problem.
+
+    References:
+        [1]: "Bin packing",
+        https://en.wikipedia.org/wiki/Bin_packing_problem
+    """
+
+    def __init__(self, weights: List[int], max_weight: int) -> None:
```

Comment:
It might be good to have `max_num_bins` as `Optional[int]`. If it is `None`, you can set the number of items as `max_num_bins` as the trivial upper bound.